### DISCUSSION TITLE
Facing issue in configuration of 8x8 mesh topology hnoc via omnet

### DISCUSSION THREAD

**Discussion by Anamt761** [omnetpp/omnetpp#980] [Q&A] [answered]
hi! I have imported the framework HNOC and run an example of async. 8x8 mesh topology 
but there is an error, I fixed it but can't work
please let me know! what should I do to run 8X8 mesh topology with XY Routing?

![112](https://github.com/omnetpp/omnetpp/assets/48387290/edcba583-4039-4a6f-9e44-f272407cc93f)

Also, I figure out the FAQ section of hnoc that the proposed solution is attached but the framework folder that is live on Github,  (https://github.com/yanivbi/HNOCS) contains no hnocs.exe file

![111](https://github.com/omnetpp/omnetpp/assets/48387290/ed941110-d791-42b9-a69f-ee8aee2d907d)

if anyone know, please guide me how to build its configuration

thanks & Regards

**Comment by rhornig** [comment, marked as answer]
That model was written for OMNeT++ 5.4, but you are trying to run it under OMNeT++ 6.0.1. There was a change regarding the default context of the submodule parameters. Previously a `parameter = parameter` assignment meant to assign the parent's `parameter` to the submodule's `parameter`. In OMNeT++ 6, this should be written as `parameter = parent.parameter` (see the 6.0 release notes).

Long story short, I suggest to follow the installation instructions and use OMNeT++ 5.4 (or 5.7 if you are more adventurous). Porting the model to 6.0 would be a bit more complex.

**Reply by Anamt761** [reply]
Thanks for the quick response, issue is resolved, One more thing please let me know if we want to configure task mapping, (NoC), then which file contains the mapping function to perform algorithm results, and trace file also. 
Thanks

**Reply by rhornig** [reply]
I'm not familiar with that model, so I cannot help on this.

**Comment by Anamt761** [comment]
how to perform application mapping with VoPD trace file through HNoC? 
I tried to perform the application VOoPD, and PIP in 8x8 async mesh topology but it didn't read the file.?
please let me know, how to give them a path as I add the file in asynch omnet.ini file.

# Source Parameters
**.source.pktVC = 0  # the VC injecting the packet on from the NI 
**.source.msgLen = 4 # packets per message
**.source.pktLen = 8 # in flits
**.source.isSynchronous = true # inject flits without any synchronization to clock
**.source.isTrace = true  # do not inject based on trace file
**.source.fileName = "task_graph_VoPD.txt"    # no trace file given
**.source.flitArrivalDelay = 2ns  # 1 flit / Cycle
**.source.maxQueuedPkts = 16
**.source.dstId = (id + intuniform(1, 63)) % 64 # Uniform random thar prevent self dst 

I made changes in such parameters
**.source.isSynchronous = false# inject flits without any synchronization to clock
**.source.isTrace = false# do not inject based on trace file
**.source.fileName = " "    # no trace file given

but didn't get the results as file cores-->source--?pktfifosrc.cc shows error due to trace file
